### PR TITLE
feat: add support for jenkins-helper git ref when build pipeline

### PR DESCRIPTION
**Pull request by zdtsw**
Fix: https://github.com/adoptium/ci-jenkins-pipelines/issues/419 

As stated in the doc: `
You may find that the `RepoHandlerTest.groovy:adoptDefaultsGetterReturns()` will fail when you add new values. This is expected as the test is pulling in Adopt's master branch `defaults.json` which does not yet contain the new values. Please inform any reviewers of this in the pull request.`
thats why GH action groovy test failed on adoptDefaultsGetterReturns because on master helper_ref does not exist as null

```
RepoHandlerTest > adoptDefaultsGetterReturns() FAILED
    org.opentest4j.AssertionFailedError: expected: <null> but was: <master>
        at org.junit.jupiter.api.AssertionUtils.fail(AssertionUtils.java:55)
        at org.junit.jupiter.api.AssertionUtils.failNotEqual(AssertionUtils.java:62)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
        at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
        at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1124)
        at org.junit.jupiter.api.Assertions$assertEquals$1.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
        at RepoHandlerTest.adoptDefaultsGetterReturns(RepoHandlerTest.groovy:27)
```

### CODE REVIEW

**Comment by zdtsw** [comment]
run tests quick 11

**Comment by zdtsw** [comment]
run tests quick 11

**Comment by zdtsw** [comment]
run tests quick 11

File: pipelines/build/common/build_base_file.groovy [hunk 1]
```diff
@@ -626,7 +627,8 @@ class Builder implements Serializable {
         try {
             context.timeout(time: pipelineTimeouts.API_REQUEST_TIMEOUT, unit: "HOURS") {
                 // Query the Adopt api to get the "tip_version"
-                def JobHelper = context.library(identifier: 'openjdk-jenkins-helper@master').JobHelper
+                String helperRef = DEFAULTS_JSON['repository']['helper_ref']
```
Comment (andrew-m-leonard) [inline pipelines/build/common/build_base_file.groovy:630]: Probably needs to check if helper_ref is blank or not set? then default to master?

Comment (andrew-m-leonard) [reply pipelines/build/common/build_base_file.groovy:630]: or can we assume DEFAULTS_JSON must set it?

Comment (zdtsw) [reply pipelines/build/common/build_base_file.groovy:630]: so i added it to the pipeline/defaults.json to master as the default value.
see test : https://ci.adoptopenjdk.net/job/build-scripts-pr-tester/job/build-test/job/openjdk11-pipeline/930/parameters/

**Comment by karianna** [comment]
@zdtsw Should we expect the Test / Groovy GitHub action to pass on this?

**Comment by zdtsw** [comment]
> @zdtsw Should we expect the Test / Groovy GitHub action to pass on this?

GH action test is based on master's defaults.json which does not have the new helper_ref yet, so the test is expected to get "null" than "master"
once this is updated, test should be fine.

**Comment by karianna** [comment]
@zdtsw Still seems to be failing the test and the linter...

**Comment by zdtsw** [comment]
close this PR, will add function into https://github.com/adoptium/ci-jenkins-pipelines/pull/423

**Pull request closed without merging**